StortSpelProjekt/Engine/src/Misc/AssetLoader.cpp

```diff
@@ -129,6 +135,214 @@ Model* AssetLoader::LoadModel(const std::wstring& path)
 	return m_LoadedModels[path].second;
 }
 
+HeightmapModel* AssetLoader::LoadHeightmap(const std::wstring& path)
+{
+	// Check if the heightmap model already exists
+	if (m_LoadedModels.count(path) != 0)
+	{
+		return dynamic_cast<HeightmapModel*>(m_LoadedModels[path].second);
```

Comment:
static_cast

Reply:
If the loaded model is a Model and not a HeightmapModel I don't want to return it as the user expects a heightmap. An extra check and log could be added for this reason though.